docs/designs/operator.md

```diff
@@ -0,0 +1,201 @@
+# adx-mon Operator Design
+
+## Overview
+
+The `adx-mon-operator` is a Kubernetes operator responsible for managing the lifecycle of an adx-mon cluster, including its core components (collector, ingestor, alerter) and required Azure infrastructure. The operator leverages [Azure Service Operator (ASO)](https://azure.github.io/azure-service-operator/) to declaratively manage Azure resources such as ADX (Azure Data Explorer) clusters, databases, and other dependencies.
+
+The operator aims to provide a simple, production-ready bootstrap experience for adx-mon clusters, while supporting advanced customization for complex and federated deployments.
+
+---
+
+## Responsibilities
+
+- **Azure Infrastructure Management:**  
+  - Declaratively create and manage Azure resources (e.g., ADX clusters) using ASO CRDs.
+  - Wait for ASO-managed resources to reach a ready state before proceeding with dependent adx-mon components.
+
+- **adx-mon Component Management:**  
+  - Generate and apply manifests for adx-mon components (collector, ingestor, alerter) as Kubernetes resources (e.g., StatefulSets, Deployments, Services).
+  - Support default images for each component, with the ability to override via CRD spec.
+  - Allow configuration of the number of ingestor instances (replicas).
+  - Support further customization of component manifests via CRD fields.
+  - Support deployment of only a subset of components (e.g., only collectors, or only ingestors) to enable federated/multi-cluster topologies.
+
+- **Reconciliation and Drift Detection:**  
+  - Watch the manifests for adx-mon components.
+  - If a user manually changes a managed resource (e.g., increases ingestor replicas), detect the change and update the operator CRD's status/spec to reflect the new state.
```

Comment:
re: scaling ingestor... that is something that the operator should be able to handle.  Just thinking out loud, but we might want a `manual` mode as well as an `auto-scaling` mode to allow a user to control the scaling if they want.